### ISSUE TITLE
Update LOOT support for importing Bash Tags

### ISSUE THREAD

**Issue by Sharlikran** [wrye-bash/wrye-bash#299]
### Issue:

LOOT is still being updated and maintained by WrinklyNinja.  While not part of the Wrye team the interaction between LOOT and Wrye bash is essential.  This critical integration provides the importing of Bash Tags from the LOOT masterlist.yaml, userlist.yaml, and a taglist.yaml file provided with Wrye for Oblivion and Skyrim.  

BOSS support is no longer included with Wrye and would be a waste as the only list maintained is for Oblivion.  All other master lists are locked at this time.
### Request:

Update Wrye to use the current snaphot of the LOOT API

Currently 0.6.0 of the API is in use by Wrye and is outdated.  It can't read the new LOOT syntax.  The most recent working snapshot that will work with 0.8.1 of LOOT is API v0.8.1-5-g0ba3dae.  It has no major change to the API and can be used with Wrye without changes to existing code.

The current [loot-api_0.9.1-85-g2972478_dev.7z](https://bintray.com/wrinklyninja/loot/LOOT_API#files) has had breaking [changes in the API](https://github.com/loot/loot/commits/dev/include/loot) and cannot be used without updating existing code.  The changes may not be significant and there aren't any behavioral changes.

If it helps you could always compare the PDF provided with [LOOT_API_v0.8.1-5-g0ba3dae](http://www.mediafire.com/download/z22gqq6lc7n2h8d/LOOT_API_v0.8.1-5-g0ba3dae.7z) with the new PDF in loot-api_0.9.1-85-g2972478_dev.7z and see what has changed.

The older spapshot for 0.8.1 listed doesn't include support for Fallout 4.  While the latest release does.
### Impact:

There is no impact for users of Oblivion and Skryim if the LOOT is API v0.8.1-5-g0ba3dae is used unless there are more syntax changes in the future that require an update.  However, without Fallout 4 support there would be no way to import Bash Tags for Fallout 4.

Currently no patchers for Fallout 4 are active but the first one and I believe the minimal that should be provided is the Leveld List patcher.  While users can add bash tags manually and will need to do so initially, users really have no idea when ReLev and DeLev need to be applied.  While adding ReLev and DeLev to every plugin will make Wrye cringe and simply not change something it will not produce a correct Bash Patch in the end.  Users are not advised to add tags for no reason.
### Suggested method for troubleshooting and resolution:
1. Run Wrye Bash with db99d2559ab8f5f852a8e26ba0fa52dc0a60c678
2. Click on a plugin and look at the suggested tags in the Bash tags box located in the lower right corner.
3. Open the taglist.yaml file or masterlist.yaml file from LOOT for oblivion and compare the tags.

Review [changes in the API](https://github.com/loot/loot/commits/dev/include/loot) (also linked above) or the PDF provided with the LOOT API and make the necessary changes in loot.py until the Bash Tags are displayed correctly.

**Comment by Sharlikran** [comment]
@lojack5 @valda I know the two of you are busy and I know Utumno can update this when he has time.  However, I am wondering if either of you could look at this and make the proper updates.

The best way to test this is to simply use it with Oblivion.  What happens is that LOOT provides bast tags so differently Wrye Bash ends up adding tags that don't exist in the yaml files.

If you use it with Skyrim there are some pretty obvious changes that occur as [reported by Alt3rn1ty](http://forums.bethsoft.com/topic/1599816-wrye-bash-thread-108/?p=25144776) in the Bethesda forums.  For example adding the Sounds tag to the empty dummy plugins for the High Res Texture packs.

**Comment by Sharlikran** [comment]
@iaz3 also if you happen to have time the help would be appreciated.

**Comment by Sharlikran** [comment]
@acui I know it has been a long time since we spoke last, but I am wondering if you have any time and would be willing to help with this.  You could use Oblivion or Skyrim to test this if you don't have Fallout 4.

**Comment by Sharlikran** [comment]
And just to be completely through and bring everyone out of the woodwork, would @D4id4los or @jfpelland be willing to help out if possible.  :+1:

**Comment by Utumno** [comment]
See: http://forums.bethsoft.com/topic/1602198-wrye-bash-thread-109/#entry25144990

**Comment by Utumno** [comment]
I added https://github.com/wrye-bash/wrye-bash/pull/310/commits/db99d2559ab8f5f852a8e26ba0fa52dc0a60c678 to my branch and was greeted in oblivion with:

![oblivion](https://cloud.githubusercontent.com/assets/1365079/15561087/aa7fd206-22f2-11e6-9225-d17a48c4c2cb.jpg)

Is this the oblivion masterlist issue you were referring to ?

**Comment by Sharlikran** [comment]
After asking WrinklyNinja the error was a bug in the API mostly.  Which is disappointing because I mentioned it to him at one point.  However, I probably didn't think it was a bug with the LOOT.  I presented it as an issue with the Python code and how it uses the LOOT API.  Which is probably why he didn't think it was a bug with LOOT either.  I will be enabling the Leveled List patcher for Fallout 4 and getting it up and running just no ETA on that.

**Comment by Utumno** [comment]
Yep I know - I posted a bug since and was fixed - in my utumno-wip I bundle loot 9.2 that does not require the dll (that's loot 10) - you can use that.

**Comment by Sharlikran** [comment]
Nah, the dll required is documented.  I already have one of the snapshots in use https://github.com/Sharlikran/wrye-bash/commit/78cd40703e317a8435743d88ba038fef90d89569 and it's working just fine.

**Comment by Utumno** [comment]
Maybe it's required for your snapshot ? The new API @WrinklyNinja built definitively requires the dll and I asked him to provide some help integrating this in Bash - see: https://github.com/wrye-bash/wrye-bash/issues/325#issuecomment-247876454

Depending on the help I get I may integrate this (loot 10) - otherwise loot 9. I just want to avoid binary commits every so often it's a cancer for a repository.

As you see I eventually came round to it as I would - I fully understand the significance of it but as seen even loot was not ready at that point, let alone Bash.

I need trust, or code ;)

----

Btw the utumno-lo-rc3 is ancient (and buggy) history. I would advice you to clean up your fork and kindly rework the links in this repo to point to your rebased branches. I will put the loot commit in the base of my branch so you can rebase on that so we use the same loot.

When/if you can

**Comment by WrinklyNinja** [comment]
Since I've been mentioned, I'll just pop in to say that I converted the mktaglist.py script to use the new LOOT API as a warm-up, and I'll take a look at the actual Bash usage and the NSIS installer soon (TM). I want to try improving LOOT's exception implementation in parallel though, this will also help mapping to Python exceptions.

**Comment by Utumno** [comment]
Good to know - wrinlkly I was a bit annoyed by copy paste code in mktaglist so I kind of reduced it :whistling: - I hope you used the dev version. Which will be probably further reduced - no use querying the registry, as the dev using it may have not all the games installed :whistling:

**Comment by Sharlikran** [comment]
@Utumno I wasn't able to rebase feature-sharlikran-valda-fallout on utumno-wip because of the changes to mod_links.py.  With the new syntax I would have no idea how to add the FidListMerger in commit https://github.com/Sharlikran/wrye-bash/commit/4398d7f84266beae1c2b2e07b651d0ff1bbce97b.  

When you have time can you rebase feature-sharlikran-valda-fallout on utumno-wip so I can look at the changes and see how things work now?

I didn't rebase off of dev because there is too much in utumno-wip that I would want merged into dev.  I could wait I guess until you come to a stopping point and have time to merge most of utumno-wip into dev.

I just don't want interfere with your current plans and limited time.  Not sure how to proceed.

**Comment by Utumno** [comment]
Hey @Sharlikran - thanks for trying :)

I am actively merging my wip branch to dev but it's a difficult process, as always (much easier/fun to commit than rearrange, rethink stuff, rethink again, rewrite messages to explain what's in the commit (duh) and above all _review_ and make sure nothing breaks, yep that's the hard part).

Will rebase your branch asap - but meanwhile I have rebased already on an older version of dev here:
https://github.com/wrye-bash/wrye-bash/tree/feature-sharlikran-valda-fallout - have a look (or did you try to rebase that one ?)

Was planning to rebase when I merge the patchers stuff from my wip - I hope I will be able to do this in the week end (will suck up all the week end)

Well, coffee...

**Comment by Sharlikran** [comment]
I looked at it but utumno-wip has more changes and fixes.  I might take another look and see if can figure out whether or not you just moved the code.  If not I can wait.  I have a working branch that does what I need for now and I can still use it to set up the leveled list patcher.

**Comment by Utumno** [comment]
The commit that bugs you is 743eee48a5dc1ba64e761789de338e7d50a2be9f - I introduced a log_config method in the _PathcherPanel hierarchy and instead of checking isinstance(ThisOrThatPanel) I moved the code in log_config overrides. But as I see the commit you linked me to has also other problems - better use my rebased branch or better wait till I rebase myself. The problems are related to getConfig. What I basically set up to do in this phase of refactoring is move the config logic to the gui side (gui_patchers.py). It complicated the already complex logic of the patchers - the patchers should have nothing to do with lousy checkboxes and the like that belong to the gui. I am almost done which means the patchers will be now easier to work with. I intend to merge these changes on dev but meanwhile please use my rebased branch as it contains already some fixes - so for the FidPatcher here is the fixed commit in the rebased branch - see I moved getConfig. Actually it would be grand if you could checkout this branch and test all is working fine - so I can then proceed to rebase it on the new dev with more confidence. I don't have fallout3,4,nv so I just edit the code. I am 99.9% sure it works ok and has all your changes (except new ones you made since) but it would be really helpful to test it.

**Comment by Utumno** [comment]
Just to add that I am now finalizing the #259 related code (a _huge_ pain and a huge mess in renames finally fixed) and then I will proceed adding the patchers commits on dev - so avoid rebasing on my branch or on dev as it will change often in the next few days. Just testing my rebased branch will be a huge help

**Comment by Sharlikran** [comment]
I think it's probably best for me to just wait so I don't interfere with what limited time you have.  Thanks for pointing me to the commits that changed the code.

**Comment by Utumno** [comment]
Don't worry it'll be done soon - but testing my rebased branch will be a huge help - as I already fixed stuff there and this is the branch I will rebase on dev anyway. Btw I don't know if you saw it but wrinkly issued a pull request here with the latest version of his API I am eager to throw into my new branchy

**Comment by Sharlikran** [comment]
Not to drag this out too much but if you did use it as is for now, which branch would you put it in temporarily, your utumno-wip branch?

**Comment by Utumno** [comment]
Yep - will do ASAP - on my next push of utumno wip - actually I look forward to packing a new exe for users to test this loot 10 thingy :)

I still fight with renames agggr

**Comment by Sharlikran** [comment]
I have been busy with the fam at the moment but I am pretty sure I could rebase everything except the FO3/FNV stuff.  So I'm gonna try that real quick.

**Comment by Sharlikran** [comment]
Not 100% sure but I think I have my LOOT update branch properly rebased on utumno-wip.  Can you guys @WrinklyNinja @Utumno see what you think?  

Branch: https://github.com/Sharlikran/wrye-bash/tree/dev-skyrim-patcher-loot-update

If it's wrong let me know and I want to fix it for more practice.

**Comment by WrinklyNinja** [comment]
Well, my commits seem to have been applied correctly.

**Comment by Sharlikran** [comment]
Thanks for checking that I got all the commits from your pull request correctly.  I'm still not an expert at all this git stuff.  I appreciate it.

**Comment by Utumno** [comment]
Closing this as it's superseded by #329 - comment there for loot related stuff (especially if we are to bundle it)
